parachain/pallets/incentivized-channel/src/outbound/mod.rs

```diff
@@ -216,52 +222,49 @@ pub mod pallet {
 			let next_nonce = nonce.saturating_add(1);
 			<Nonce<T>>::put(next_nonce);
 
-			let bundle =
-				MessageBundle { nonce: next_nonce, messages: messages.clone().into_inner() };
+			let bundle: MessageBundleOf<T> = MessageBundle {
+				nonce: next_nonce,
+				fee: (messages.len() as u128).saturating_mul(Self::fee()),
```

Comment:
hmm i'm just thinking about if anything has changed in terms of assumptions for how different messages depend on eachother. the goal of the design has always been to ensure that there is no dependency between messages, ie, if a message is broken or not dispatchable/callable or whatever, it does not affect other messages.

fees are fixed, and completely controlled and consumed by our channel logic with application-specific messages having no say/involvement in the fees, right? so this means we can roll them up and retain the no-dependency-between-messages goal? this all sounds good - will you just think about this yourself a bit more too and clarify/confirm 100% that this remains the case?

The other thing to keep in mind is that if we have different kind of channels in future that are not fixed-fee, and rather based on something like a fee market where applications can choose their fee, then this bundling interface won't be able to support that easily. Anyway, even though I mention that, it's not really an issue as would be dealt with in that later feature work anyway.

👍👍

Reply:
Hey yeah, I did think about these considerations. Indeed one of primary motivators of this change was to enable support for more flexible fee models (fee markets, etc). So we can introduce fee models of varying complexity without having to upgrade our wire protocol every time. 

Firstly, when dealing with fees, we can take the analogy of stock market broker. The front office (equivalent to the outbound channel on the parachain), deals with fee charging according to market dynamics. The back office (equivalent to the inbound channel on ethereum) deals with settlement of rolled up fees to a bank account. The back office doesn't need to know that a specific message incurred a certain fee. In essence, back office only needs to know that a specific account needs to credited a certain amount. 

So in essence, I've decoupled the fee market from final fee settlement.

Reply:
right though if we have a fee market, the ordering+inclusion of messages would be determined by the destination chain logic, not the source chain logic, so I don't think that would fit into this design

Reply:
I know this discussion is getting a bit off-topic, but I'm keen to understand more about the details of the fee market you have in mind. :)

My understanding was that message inclusion & ordering (based on fee markets) would be determined by the source chain, since its in the best position to buffer messages, extract fees from users, and interact with Oracles if necessary. 

The destination chain could safely reorder the included messages I guess, but then again the source chain could do that too.

Reply:
ordering on source chain is nice, but is always vulnerable to the risk that there is a gas price spike such that the fees banked are not enough to cover dispatch on the destination chain, and so messages won't be dispatched until the dispatch becomes affordable again with a gas price recovery.

this is acceptable and has graceful recovery in our current design for most real world scenarios, but a theoretical example in which gas price spikes up within a very short time window and never comes back down again would halt the bridge until some relayer accepts taking a loss to flush the channel.

a more theoretically pedantic design would need ordering&inclusion to occur on the destination chain to prevent the above edge case.